### ISSUE TITLE
Linking to studio palette does nothing

### ISSUE THREAD

**Issue by Quargzon** [tahoma2d/tahoma2d#564]
**Describe the bug**
After creating a studio palette and using it in one or more levels, modifying the studio palette and saving (overwriting) it does not appear to update linked colors/styles, even if Tahoma2D is restarted and the project is reloaded entirely.

It's worth noting that I am a fairly new user, so I may not be using the feature correctly. Also, if this is indeed a bug, I don't know exactly how it's supposed to behave when it's working as expected, so I apologize for any confusion.

**To Reproduce**
Overview:
1. Create a studio palette with some colors
2. Load the studio palette onto a level and draw something with the colors in it
3. Make sure the colors are linked to the studio palette (black arrows in the little white boxes on each color)
4. Draw something with the colors
5. Change the colors in the studio palette and save it
6. The changes to the studio palette are NOT reflected in the levels that use it, even after restarting/reloading

Detailed steps:
1. Create a new scene
2. Go to Studio Palette panel -> right click "Project Palettes" -> new palette
3. Add some different colors to the newly created studio palette
4. Click the save button on the lower-right of the studio palette panel
5. Add a **vector** level (I don't know if this happens with raster since I haven't tried, but I get the problem with vector)
6. Drag the new palette onto the small grid button on the lower-right of the **level** palette panel to load it
7. Select all the palette colors -> right click one of the colors -> Toggle Link to Studio Palette
8. Draw something with these colors
9. Modify the colors in the **STUDIO** palette from the **STUDIO** palette panel
10. Click the save button on the studio palette panel again and choose "overwrite"
11. None of the colors update to match the studio palette
12. (optional) save the scene, restart Tahoma2D, and reload the project -- still no change

**Device Information:**
 - OS: Windows 10
 - Tahoma2D Version: 1.1
 - GPU: NVIDIA GTX 1070
 - GPU Driver Version: 460.89

**Comment by Quargzon** [comment]
Also, this appears to be mentioned in the comments of a [PR on OpenToonz](https://github.com/opentoonz/opentoonz/pull/628) I just dug up from back in 2016. The comments seem to address this issue but then trail off inconclusively without any solution.

**Comment by Quargzon** [comment]
I'm not familiar enough with this application or codebase to make much sense of the code, but after some sleuthing, it looks like `StudioPaletteCmd::updateAllLinkedStyles()` defined in `toonz/sources/toonz/studiopalettecmd.cpp` is at the heart of the issue.

Probably either `updateAllLinkedStyles()` isn't getting called when a studio palette is modified for some reason, or something about the implementation of `updateAllLinkedStyles()` doesn't work as expected in that particular case.

`StudioPalette::updateLinkedColors()` defined in `toonz/sources/toonzlib/studiopalette.cpp` seems to compare palette color _name strings_ to decide how to update the styles, so maybe the internal naming gets messed up somewhere along the way.

This is just blind speculation on my part, but I hope it can save some time for someone who knows what they're doing!

**Comment by artisteacher** [comment]
It's working just fine for me on macOS. What level type are you using? Vector levels and smart raster levels are color mapped and should update when the color is changed. Standard raster level drawings are not color mapped and won't change color.

**Comment by Quargzon** [comment]
I'm using vector levels (on Windows). I've never been able to get a drawing to update its colors based on a studio palette so far. The only way I've found is to directly go through each level and use "Load into Current Palette".

In that other PR, I saw people saying things like:
> Yet I did something to get it to work. What am I missing

> I will write down if i manage to figure out how I got linking to work once.

I'm wondering if maybe there's an ultra-specific way you have to set things up that just happens to work, but otherwise linking is broken... What steps exactly do you do to get it working? For example, how are you loading the studio palette into the level initially **BEFORE** drawing? Do you drag it into the palette window? Or the little grid icon in the corner? Or do you use "Load into Current Palette" or something else?

**Comment by Quargzon** [comment]
This issue can be considered closed as it doesn't seem to be a bug, but rather a deliberate but confusing "feature". I just want to leave it open a bit longer in case a dev wants to make it a feature request instead.

**Comment by Quargzon** [comment]
OK, I'm still pretty confused, because I was trying apply the "solution" I found to an existing project and it didn't fix anything. Eventually I realized that if there are big black X's through the colors in the palette, it means there's something wrong with the link (not entirely sure about this). All I know is that when I observed the studio palette link working, the X's weren't there.

In any case, I think that means I can safely say this wasn't a bug, but rather that something in my project got messed up somehow and broke the link. Closing this issue.
